src/tests/nativeaot/CustomMain/CustomMain.csproj

```diff
@@ -3,6 +3,7 @@
     <OutputType>Exe</OutputType>
     <CustomNativeMain>true</CustomNativeMain>
     <StaticLibraryPrefix Condition="'$(TargetOS)' != 'windows'">lib</StaticLibraryPrefix>
+    <NativeAotIncompatible Condition="'$(TargetArchitecture)' == 'arm64' and '$(TargetOS)' != 'windows'">true</NativeAotIncompatible>
```

Comment:
I was pretty sure the test was supposed to check an osx-arm64 scenario so it feels wrong to disable that.

Reply:
I was fighting with cross-building the dependencies when trying to run something on maccatalyst-arm64/x64 but it definitely was not broken for the osx-* combinations that I tried.

At minimum, can the condition be relaxed to allow running the test when the scenario is not cross-compilation?